arangod/RestHandler/RestStatusHandler.cpp

```diff
@@ -274,3 +285,27 @@ RestStatus RestStatusHandler::executeOverview() {
   generateResult(rest::ResponseCode::OK, result.slice());
   return RestStatus::DONE;
 }
+
+RestStatus RestStatusHandler::executeMemoryProfile() {
+#if defined(USE_MEMORY_PROFILE)
+  bool found;
+  std::string outDirectory = _request->value("directory", found);
+
+  if (!found) {
+    outDirectory = TRI_GetTempPath();
+  }
+
+  std::string const fileName = FileUtils::buildFilename(outDirectory, "profile.out");
+  char const* f = fileName.c_str();
+  mallctl("prof.dump", NULL, NULL, &f, sizeof(const char *));
+  std::string const content = FileUtils::slurp(fileName);
+  TRI_UnlinkFile(f);
```

Comment:
I think allowing to specify arbitrary directories names via URL parameters is a security issue.
In addition, I am not sure what happens if the target directory does not exist. Will the memory profile dump fail?
This is also not thread-safe, because if no directory is specified, multiple callers will write to the same output file concurrently.

Instead of all this, we should also create a temporary filename in the temporary directory, write the dump into it and then remove it again.

Last but not least, the profile dump file is not cleaned up if an exception occurs in FileUtils::slurp(...). We should clean up the profile dump file in all cases, so I guess we should remove it via a scopeGuard.

Reply:
done